src/modules/rc_update/RCUpdateTest.cpp

```diff
@@ -131,3 +186,60 @@ TEST(RCUpdateTest, ModeSlotButtonAllValues)
 	checkModeSlotButton(31, 6, 1.f, 0); // button 6 pressed but not configured -> manual_control_switches_s::MODE_SLOT_NONE
 	checkModeSlotButton(63, 6, 1.f, 6); // button 6 pressed -> manual_control_switches_s::MODE_SLOT_6
 }
+
+TEST_F(RCUpdateTest, ReturnSwitchUnassigned)
+{
+	// GIVEN: Default configuration with no assigned return switch
+	EXPECT_EQ(_param_rc_map_return_sw.get(), 0);
+
+	// WHEN: we update the switches two times to pass the simple outlier protection
+	_rc_update.UpdateManualSwitches(0);
+	_rc_update.UpdateManualSwitches(0);
+
+	// THEN: we receive an unmapped return switch state
+	uORB::SubscriptionData<manual_control_switches_s> manual_control_switches_sub{ORB_ID(manual_control_switches)};
+	manual_control_switches_sub.update();
+
+	EXPECT_EQ(manual_control_switches_sub.get().return_switch, 0); // manual_control_switches_s::SWITCH_POS_NONE
+}
+
+TEST_F(RCUpdateTest, ReturnSwitchPositiveThresholds)
+{
+	checkReturnSwitch(-1.f, 0.5f, 3); // Below threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(0.f, 0.5f, 3); // On threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(.001f, 0.5f, 1); // Slightly above threshold -> SWITCH_POS_ON
+	checkReturnSwitch(1.f, 0.5f, 1); // Above threshold -> SWITCH_POS_ON
+
+	checkReturnSwitch(-1.f, 0.75f, 3); // Below threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(0.f, 0.75f, 3); // Below threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(.5f, 0.75f, 3); // On threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(.501f, 0.75f, 1); // Slightly above threshold -> SWITCH_POS_ON
+	checkReturnSwitch(1.f, 0.75f, 1); // Above threshold -> SWITCH_POS_ON
+
+	checkReturnSwitch(-1.f, 0.f, 3); // On minimum threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(-.999f, 0.f, 1); // Slightly above minimum threshold -> SWITCH_POS_ON
+	checkReturnSwitch(1.f, 0.f, 1); // Above minimum threshold -> SWITCH_POS_ON
+
+	checkReturnSwitch(-1.f, 1.f, 3); // Below maximum threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(1.f, 1.f, 3); // On maximum threshold -> SWITCH_POS_OFF
+}
+
+TEST_F(RCUpdateTest, ReturnSwitchNegativeThresholds)
+{
+	checkReturnSwitch(1.f, -0.5f, 3); // Above threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(0.f, -0.5f, 3); // On threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(-.001f, -0.5f, 1); // Slightly below threshold -> SWITCH_POS_ON
+	checkReturnSwitch(-1.f, -0.5f, 1); // Below threshold -> SWITCH_POS_ON
+
+	checkReturnSwitch(1.f, -0.75f, 3); // Above threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(.5f, -0.75f, 3); // On threshold -> SWITCH_POS_OFF
+	checkReturnSwitch(-.001f, -0.75f, 1); // Slightly below threshold -> SWITCH_POS_ON
```

Comment:
```suggestion
	checkReturnSwitch(.499f, -0.75f, 1); // Slightly below threshold -> SWITCH_POS_ON
```

I think in general it's quite confusing having two different scales (-1 ... +1 for channel values & 0 ... 1 for threshold), but anyhow this value needs to be changed 🤔

Reply:
Right you are! Thanks for looking into the details 🙏